test/parallel/test-zlib-truncated.js

```diff
@@ -45,5 +45,22 @@ const inputString = 'ΩΩLorem ipsum dolor sit amet, consectetur adipiscing el'
     zlib[methods.decomp](truncated, function(err, result) {
       assert(/unexpected end of file/.test(err.message));
     });
+
+    const syncFlushOpt = { finishFlush: zlib.Z_SYNC_FLUSH };
+
+    // sync truncated input test, finishFlush = Z_SYNC_FLUSH
+    assert.doesNotThrow(function() {
```

Comment:
(Just a suggestion/ idea): Maybe you could add a case that triggers the new error if the flush flag is not valid? (Just to make sure that the flag was caught somewhere)